src/network/networkd-link.c

```diff
@@ -2664,12 +2729,25 @@ int link_update(Link *link, sd_netlink_message *m) {
                 r = link_add(manager, m, &link);
                 if (r < 0)
                         return r;
+
+                (void) link_remove_by_name(link, true, false);
```

Comment:
this doesn't look right, since you remove the entries for the new name here, but you should remove them for the *old* name

Reply:
yeah right. I moved it up before link_drop.